### PR TITLE
Update animal records

### PR DESCRIPTION
**Pull request by duskotmane**
# In this PR : 

- [ ] I Add a column species of type string to the animal's table.
- [ ] Insert the all-new data.
- [ ] Update the animals table by setting the species column to digimon or pokimon.
- [ ] Delete all animals born after Jan 1st, 2022. 
- [ ] Update all animals' weights that are negative to be their weight multiplied by -1.
- [ ] Write queries to answer all the asked questions.

- Here are the screenshots : 

fordelete
makepositive
q1
q2
q3
q4
q5
secRev
lst

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -5,4 +5,61 @@ SELECT date_of_birth FROM animals WHERE name = 'Agumon' OR name = 'Pikachu';
...
+WHERE name LIKE '%mon';
+
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL OR species = '';
+SELECT * FROM animals;
+
+COMMIT;
+SELECT * FROM animals;
+
```
Comment (Gambit142) [inline queries.sql:33]: - [ ] Before the transaction highlighted in the code snippet, you ought to write a transaction that sets all `species` column to `'unspecified'` and then roll it back. In the process of the transaction, you ought to verify that the columns were set to `'unspecified'` by selecting all from the `animals` table. Kindly carry out this transaction as it is a requirement. 👍 💻 

image

### N:B: Please update the screenshots in the pull request summary with the verified changes of this transaction. 😎

File: queries.sql [hunk 2]
```diff
@@ -5,4 +5,61 @@ SELECT date_of_birth FROM animals WHERE name = 'Agumon' OR name = 'Pikachu';
...
+SET weight_kg  = weight_kg  * -1;
+
+ROLLBACK TO SAVEPOINT first_save;
+
+UPDATE animals
+SET weight_kg = weight_kg * -1
+WHERE weight_kg < 0;
+
+COMMIT;
+SELECT * FROM animals;
```
Comment (Gambit142) [inline queries.sql:60]: - [ ] Before the transaction highlighted in the code snippet, you ought to write a transaction that deletes all data in the `animals` table and then roll it back. In the process of the transaction, you ought to verify that all data were deleted from the `animals` table. Kindly carry out this transaction as it is a requirement. 👍 💻 

image

### N:B: You have the screenshot attached, but the code is not written in this file. Kindly write the code for it.  😎

File: queries.sql [hunk 3]
```diff
@@ -5,4 +5,61 @@ SELECT date_of_birth FROM animals WHERE name = 'Agumon' OR name = 'Pikachu';
...
+GROUP BY neutered;
+
+SELECT neutered, MIN(weight_kg ), MAX(weight_kg)
+FROM animals
+GROUP BY neutered;
+
+SELECT species, AVG(escape_attempts) AS average_escape_attempts
+FROM animals
+WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31'
+GROUP BY species;
```
Comment (Gambit142) [inline queries.sql:81]: - [ ] The screenshot you attached for this query displays the wrong result. The correct result should be `3.0`. If you look at the data in the `animals` table, only `Blossom` and `Squirtle` are born between _**1990 to 2000**_. They both have an `escape_attempt` of `3`. Therefore, the result that should be returned from this query is `3`. Kindly attach the correct screenshot. 👍 😎

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @otmaneechchafyky 👋, 

Good job so far!

## Highlights
- Everything that creates or changes the structure of the table is in the schema.sql file.
- Everything that populates tables with data is in the data.sql file.
- Everything that is a query that helps to analyze data is in the queries.sql file.

You still need to work on some issues to go to the next project, but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

You can also consider the following:

N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by victorokoroji — APPROVED** [review]
Hi @otmaneechchafyky , 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
congratulations!  🎉💯🌟 

## Top highlight 🎯

- Requested change implemented ✅ 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**